Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -0,0 +1,383 @@
+  // Sort for lower_bound.
+  sort(current.begin(), current.end(),
+       [](const Diff& v1, const Diff& v2) { return (v1.symbol < v2.symbol); });
+
+  // If both lists are empty, write and skip.
+  if (!isize && !xsize)
+  {
+    if (include)
+      m_include = current;
+    else
+      m_exclude = current;
+    return;
+  }
+
+  // Update exclude list. Compare to exclude list if it exists and make current_diff to check
+  // against include.
+  if (!xsize && !include)
+  {
+    m_exclude = current;
+  }
+  else if (xsize)
+  {
+    for (auto& iter : current)
+    {
+      auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol, AddrOP);
+
+      if (pos->symbol != iter.symbol)
+      {
+        current_diff.push_back(iter);
+
+        if (!include)
+          m_exclude.insert(pos, iter);
+      }
+    }
+    // If there is no include list, we're done.
+    if (!include && !isize)
+      return;
+  }
+  else
+  {
+    current_diff = current;
+  }
+
+  // Update include list.
+  if (include && isize)
+  {
+    // Compare include with current and remove items that aren't in both.
+    std::vector<Diff> tmp_swap;
+    for (auto& iter : m_include)
+    {
+      if (std::any_of(current_diff.begin(), current_diff.end(),
+                      [&](Diff& v) { return v.symbol == iter.symbol; }))
+        tmp_swap.push_back(iter);
+    }
+    m_include.swap(tmp_swap);
+  }
+  else if ((!isize && include) || (!xsize && !include))
+  {
+    // If both lists are about to be filled for the first time, compare full lists and remove from
+    // include list.
+    if (include)
+      m_include = current;
+
+    for (auto& list : m_exclude)
+    {
+      m_include.erase(std::remove_if(m_include.begin(), m_include.end(),
+                                     [&](Diff const& v) { return v.symbol == list.symbol; }),
+                      m_include.end());
+    }
+  }
+  else
+  {
+    // If exclude pressed, remove new exclude items from include list that match.
+    for (auto& list : current_diff)
+    {
+      m_include.erase(std::remove_if(m_include.begin(), m_include.end(),
+                                     [&](Diff& v) { return v.symbol == list.symbol; }),
+                      m_include.end());
+    }
+  }
+}
+
+void CodeDiffDialog::Update(bool include)
+{
+  // Wrap everything in a pause
+  Core::State old_state = Core::GetState();
+  if (old_state == Core::State::Running)
+    Core::SetState(Core::State::Paused);
+
+  // Main process
+  OnIncludeExclude(include);
+
+  m_matching_results_list->clear();
+
+  new QListWidgetItem(tr("Address\tHits\tSymbol"), m_matching_results_list);
+
+  for (auto& iter : m_include)
+  {
+    QString fix_sym = QString::fromStdString(iter.symbol);
+    fix_sym.replace(QStringLiteral("\t"), QStringLiteral("  "));
+
+    QString tmp_out =
+        QString::fromStdString(StringFromFormat("%08x\t%i\t", iter.addr, iter.hits)) + fix_sym;

```

Comment:
You could possibly use `QString`'s `arg` function to make this a bit cleaner?

Reply:
Am I missing something? Seems I can't use QString() in this project, 
so the line would end up looking like

QString::fromStdString(%1\t%2\t%3\t [plus whatever formatting? havent done it yet]).arg(iter.addr, iter.hits, fix_sym);
instead of:
QString(%1\t%2\t%3\t).arg(iter.addr, iter.hits, fix_sym);

Reply:
You can use `QStringLiteral`.

Reply:
Thanks. Ended up going with :
`QStringLiteral("%1\t%2\t%3").arg(iter.addr, 1, 16).arg(iter.hits).arg(fix_sym);`
I was not able to get it with only two arg calls because of the hex format needed.

Reply:
Much better, thanks!